ui/core/components/sim_header.ts

```diff
@@ -0,0 +1,296 @@
+
+    let tooltipFragment = document.createElement('fragment');
+		tooltipFragment.innerHTML = this.warningsLink.getAttribute('data-bs-title') as string;
+		let list = tooltipFragment.children[0] as HTMLElement;
+    list.innerHTML = '';
+		
+		if (activeWarnings.length == 0) {
+			this.warningsLink.parentElement?.classList?.add('hide');
+		} else {
+			this.warningsLink.parentElement?.classList?.remove('hide');
+			activeWarnings.forEach(warning => {
+        let listItem = document.createElement('li');
+        listItem.innerHTML = warning;
+        list.appendChild(listItem);
+      });
+		}
+
+    this.warningsLink.setAttribute('data-bs-title', list.outerHTML);
+    new Tooltip(this.warningsLink);
+	}
+
+  addKnownIssue(issue: string) {
+		let tooltipFragment = document.createElement('fragment');
+		tooltipFragment.innerHTML = this.knownIssuesLink.getAttribute('data-bs-title') as string;
+		let list = tooltipFragment.children[0] as HTMLElement;
+		let listItem = document.createElement('li');
+		listItem.innerHTML = issue;
+		list.appendChild(listItem);
+		this.knownIssuesLink.setAttribute('data-bs-title', list.outerHTML);
+    new Tooltip(this.knownIssuesLink);
+	}
+
+  private addBugReportLink() {
+		let bugReportFragment = document.createElement('fragment');
+		bugReportFragment.innerHTML = `
+			<a
+				href="https://github.com/wowsims/wotlk/issues/new/choose"
+				target="_blank"
+				onclick="${onclick}"
+				data-bs-toggle="tooltip"
+				data-bs-placement="bottom"
+				data-bs-title="Report a bug or<br>Request a feature"
+				data-bs-html="true"
+			>
+				<i class="fas fa-bug fa-lg"></i>
+			</a>
+		`;
+
+		let bugReportLink = bugReportFragment.children[0] as HTMLElement;
+		new Tooltip(bugReportLink);
+		this.addToolbarItem(bugReportLink);
+	}
+
+  private addDownloadBinaryLink() {
+    let downloadFragment = document.createElement('fragment');
+    downloadFragment.innerHTML = `
+      <a
+        href="https://github.com/wowsims/wotlk/releases"
+        target="_blank"
+        class="downbin"
+        data-bs-toggle="tooltip"
+      >
+        <i class="fas fa-gauge-high fa-lg"></i>
+      </a>
+    `;
+
+    let downloadBinary = downloadFragment.children[0] as HTMLElement;
+
+		if (document.location.href.includes("localhost")) {
+			fetch(document.location.protocol + "//" + document.location.host + "/version").then(resp => {
+				resp.json()
+					.then((versionInfo) => {
+						if (versionInfo.outdated == 2) {
+              downloadBinary.setAttribute('data-bs-title', 'Newer version of simulator available for download')
+							downloadBinary.classList.add('link-danger')
+              new Tooltip(downloadBinary);
+              this.addToolbarItem(downloadBinary);
+						}
+					})
+					.catch(error => {
+						console.warn('No version info found!');
+					});
+			});
+		} else {
+      downloadBinary.setAttribute('data-bs-title', 'Download simulator for faster simulating')
+      new Tooltip(downloadBinary);
```

Comment:
I personally don't feel too good about having a constructor with side-effects, but if that's how bootstrap does it, then we should just keep it IMO.

Reply:
Yeah unfortunately Bootstrap does a pass when the page loads, but because everything is rendered in via JS we have to manually initialize like this.